modules/google-maps/src/utils.js

```diff
@@ -145,8 +150,75 @@ export function getViewState(map, overlay) {
     longitude
   };
 }
+
 /* eslint-enable max-statements */
 
+/**
+ * Get the current view state
+ * @param map (google.maps.Map) - The parent Map instance
+ * @param coordinateTransformer (google.maps.CoordinateTransformer) - A CoordinateTransformer instance
+ */
+export function getViewPropsFromCoordinateTransformer(map, coordinateTransformer) {
+  const {width, height} = getMapSize(map);
+  const {
+    lat: latitude,
+    lng: longitude,
+    heading: bearing,
+    tilt: pitch,
+    zoom
+  } = coordinateTransformer.getCameraParams();
+
+  // Match Google projection matrix
+  // The view matrix altitude is 1m, however the FOV is
+  // not calculated from this, but rather is set to 25 degrees.
+  const altitude = 1;
+  const fovy = 25;
+  const aspect = width / height;
+
+  // Match depth range (crucial for correct z-sorting)
+  const near = 0.3333333432674408;
+  const far = 300000000000000;
+
+  const projectionMatrix = new Matrix4().perspective({
+    fovy: (fovy * Math.PI) / 180,
+    aspect,
+    near,
+    far
+  });
+  const focalDistance = 0.5 * projectionMatrix[5];
+
+  return {
+    width,
+    height,
+    views: [
+      new MapView({
+        id: 'google-maps-overlay-view',
+        projectionMatrix
```

Comment:
Just place projectionMatrix in viewState?

Reply:
That doesn't work as the default null projectionMatrix from the `MapView` overrides it in https://github.com/visgl/deck.gl/blob/2601f2869ffca4c48c18c880582631ac3435e5ca/modules/core/src/views/view.js#L154

Reply:
Shall we just remove the `null` fallbacks in View constructor? They don't seem to provide any value.

Reply:
I agree, the defaults seem to get in the way. I've remove the `projectionMatrix` and `viewMatrix` defaults and updated the overlay to pass the `projectionMatrix` directly as part of the `viewState`